### ISSUE TITLE
Wire generate fails in bazel project because proto library is generated on build

### ISSUE THREAD

**Issue by dmayle** [google/wire#265]
### Describe the bug
When I try to run wire in my project, generate fails because it only sees the proto files and not the golang proto lib.

### To Reproduce
1. Create simple golang bazel project with a proto library.
2. Run `wire`.
3. Wire outputs:
```
wire: /path/to/gofile.go:X:Y: could not import path/to/proto/library (invalid package name: "")
wire: generate failed
```

## Expected behavior
Some way to run wire so that we can update the project.  I've seen #221 and I'm not looking to run wire dynamically with each build, I just want a way to run wire on provider update and check it into the source.

### Version
v0.4.0

### Additional context
None

**Comment by dmayle** [comment]
I'm trying to figure out how to work around this, at least.  I've tried copying the generated files to $GOPATH/src/path/to/proto/library but I still get the same error.  Is there any way to get more detailed error messages out of wire so I can figure this out?

**Comment by zombiezen** [comment]
Not sure, this would be failing at the `golang.org/x/tools/go/packages` level. Wire's copy may be out of date, so it might just be a case of updating the dependency. @jayconrod, any ideas?

**Comment by jayconrod** [comment]
Not sure I have enough context to answer this. @dmayle so the proto package is part of your Bazel project, not Wire or anything else?

If Wire is not being run by Bazel, then the generated .pb.go files certainly need to be in the directory where they're imported.

It's hard to say more without a minimal, reproducible example. This issue is pretty abstract.

**Comment by dmayle** [comment]
@jayconrod The real issue here was that I was using a too many build tools that are incompatible with the standard go builder.

I was trying to mix proto, GRPC, the sqlc (SQL to golang compiler) and Wire together.  At the time, I was manually generating the go files from everything, checking them in, and building with bazel.

Where I ran into a problem was that the GRPC was out-of-module (in another root / go.mod), so I couldn't get Wire to see the source.

Knowing what I know now, I could have installed it as a separate module, and then Wire should have been able to see the installed code, but instead I wrote bazel rules for both sqlc and Wire (https://github.com/plezentek/rules_sqlc and https://github.com/plezentek/rules_wire) which means that I have a working build system now, without having to manually generate sources.
